kroxylicious-test-tools/src/main/java/io/kroxylicious/test/server/MockHandler.java

```diff
@@ -51,6 +56,7 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) {
 
     private void respond(ChannelHandlerContext ctx, DecodedRequestFrame<?> frame) {
         requests.add(frame);
+        ApiMessage message = messages.get(frame.apiKey());
```

Comment:
I had been wondering about something richer along the lines of a mockito argument matcher (or maybe a hamcrest matchers) so rather than `messages.get` it would be 
```
var key expectations.keySet().forEach(matcher -> matcher.matches(frame));
ApiMessage message = expectations.get(key);
```

Allowing us to have a much richer expression of conditions. One could easily provide a convince overload to make `tester. setMockResponseForApiKey` work for the common case?

Reply:
[ArgumentMatcher](https://javadoc.io/doc/org.mockito/mockito-core/latest/org/mockito/ArgumentMatchers.html)
or 
[Hamcrest matchers](https://hamcrest.org/JavaHamcrest/javadoc/2.2/)

Or we could maybe abuse assertJ to do a similar job.

Reply:
sounds good, I'll set up the internals to be a bit more flexible